sdk/core-rest/core-client/src/common.ts

```diff
@@ -0,0 +1,30 @@
+import { PipelineOptions, RawHttpHeaders, PipelineResponse } from "@azure/core-rest-pipeline";
+
+/**
+ * General options that a Rest Level Client can take
+ */
+export type ClientOptions = PipelineOptions & {
+  credentials?: {
+    scopes?: string | string[];
```

Comment:
Out of curiosity what happens if someone passes something like `scopes: ["scope1", "scope2 scope3"]`? Do we zip them all together and send the request with `"scope1 scope2 scope3"`?

Reply:
@azure/core-rest-pipeline's bearerTokenAuthenticationPolicy takes either a string or an array we just pass it as is. I believe at the end if it is a string "scope1" it would be sent out as single item array ["scope1"]